lib/split/experiment.rb

```diff
@@ -144,11 +144,13 @@ def winner
     end
 
     def has_winner?
-      !winner.nil?
+      return @has_winner if defined? @has_winner
+      @has_winner = !winner.nil?
     end
 
     def winner=(winner_name)
       redis.hset(:experiment_winner, name, winner_name.to_s)
+      @has_winner = true
```

Comment:
TIL: setter methods always return their argument, regardless of any return statement or final value